File: src/test/java/uk/me/gumbley/minimiser/pluginmanager/NormalPlugin.java
package uk.me.gumbley.minimiser.pluginmanager;

import java.util.List;

/**
 * A normal plugin for tests.
 * @author matt
 *
 */
public final class NormalPlugin extends AbstractPlugin implements Plugin {
    /**
     * {@inheritDoc}
     */
    public String getName() {
        return "Normal";
    }

    /**
     * {@inheritDoc}
     */
    public String getVersion() {
        return "1.0.1";
    }
    /**
     * {@inheritDoc}
     */
    public List<String> getApplicationContextResourcePaths() {
        return null;
    }

    /**
     * {@inheritDoc}
     */
    public void shutdown() {
    }

    /**
     * {@inheritDoc}
     */
    public String getUpdateSiteBaseURL() {
        return null;
    }

    /**
     * {@inheritDoc}
     */
    public String getDevelopersContactDetails() {
        return null;
    }

    /**
     * {@inheritDoc}
     */
    public String getLicenseDetails() {
        return null;
    }

    /**
     * {@inheritDoc}
     */
    public String getSchemaVersion() {
        return null;
    }
}

File: src/test/resources/org/devzendo/minimiser/plugin/NormalPlugin.rb
include Java

include_class 'uk.me.gumbley.minimiser.plugin.Plugin'
include_class 'uk.me.gumbley.minimiser.plugin.AbstractPlugin'

class NormalPlugin < AbstractPlugin
  def initialize() 
      @name = 'Normal'
      @version = '1.0.1'
      @schema_version = nil
    end
    
  attr_reader :name, :version, :schema_version
  def getName
    @name    
  end
  
  def getVersion
    @version    
  end
  
  def getSchemaVersion
    @schema_version    
  end
end
NormalPlugin.new
